include/openthread/platform/crypto.h

```diff
@@ -170,8 +173,9 @@ typedef struct otPlatCryptoSha256Hash otPlatCryptoSha256Hash;
  */
 typedef struct otPlatCryptoEcdsaKeyPair
 {
-    uint8_t mDerBytes[OT_CRYPTO_ECDSA_MAX_DER_SIZE];
-    uint8_t mDerLength;
+    uint8_t  mDerBytes[OT_CRYPTO_ECDSA_MAX_DER_SIZE]; ///< Used when using literal keys, needs mKeyRef to be 0.
+    uint8_t  mDerLength;                              ///< Length of the stored DER.
+    uint32_t mKeyRef; ///< The PSA key ref (used with OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE).
```

Comment:
I think it's worth extending the documentation of `otPlatCryptoEcdsaGenerateKey` after adding this field. In particular, should we assume that `mKeyRef` is always non-zero when `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` is set and that the generated key pair is persistent?

Reply:
Makes sense. I will push a update to documentation. and it was my assumption that we use `mKeyRef` if `OPENTHREAD_CONFIG_PLATFORM_KEY_REFERENCES_ENABLE` is set. This reduces the complexity a lot.

Reply:
@Damian-Nordic Can you review the code again. I have updated the Init method to accept the keyref to be used and SRP client now provides the keyref.